### PR TITLE
remove redundant `print` method for `Splat`

### PR DESCRIPTION
**Pull request by matthias314**
I think that `print` is redundant here. The docstring for `print` says:

> `print` falls back to calling `show`, so most types should just define `show`.

### CODE REVIEW

File: base/operators.jl [hunk 1]
```diff
@@ -1339,8 +1339,7 @@ struct Splat{F} <: Function
     Splat(f) = new{Core.Typeof(f)}(f)
 end
 (s::Splat)(args) = s.f(args...)
-print(io::IO, s::Splat) = print(io, "splat(", s.f, ')')
-show(io::IO, s::Splat) = print(io, s)
+show(io::IO, s::Splat) = print(io, "splat(", s.f, ')')
```
Comment (vtjnash) [inline base/operators.jl:1342]: also a related bugfix:
```suggestion
show(io::IO, s::Splat) = (print(io, "splat("); show(io, s.f); print(io, ")"))
```

**Comment by matthias314** [comment]
Thanks! Shall I combine both commits into one?

**Comment by jishnub** [comment]
That's not necessary, as PR commits are usually squashed when they're merged.

**Comment by LilithHafner** [comment]
Thanks!

**Pull request merged**